controllers/release/adapter_test.go

```diff
@@ -656,26 +646,32 @@ var _ = Describe("Release adapter", Ordered, func() {
 			Expect(adapter.release.HasReleaseFinished()).To(BeTrue())
 		})
 
-		It("should stop reconcile if the ReleasePlanAdmission is found to be disabled", func() {
+		It("should requeue if validateProcessingResources fails with error", func() {
+			adapter.release.Labels = map[string]string{
+				metadata.AuthorLabel: "user",
+			}
 			adapter.ctx = toolkit.GetMockedContext(ctx, []toolkit.MockData{
 				{
-					ContextKey: loader.ReleasePlanAdmissionContextKey,
-					Err:        fmt.Errorf("auto-release label set to false"),
+					ContextKey: loader.ProcessingResourcesContextKey,
+					Err:        fmt.Errorf("internal error"),
+					Resource: &loader.ProcessingResources{
+						ReleasePlanAdmission: releasePlanAdmission,
+					},
 				},
 			})
 
 			result, err := adapter.EnsureReleaseIsValid()
-			Expect(!result.RequeueRequest && result.CancelRequest).To(BeTrue())
-			Expect(err).NotTo(HaveOccurred())
+			Expect(result.RequeueRequest && !result.CancelRequest).To(BeTrue())
+			Expect(err).To(HaveOccurred())
 			Expect(adapter.release.IsValid()).To(BeFalse())
-			Expect(adapter.release.HasReleaseFinished()).To(BeTrue())
+			Expect(adapter.release.HasReleaseFinished()).To(BeFalse())
 		})
 
-		It("should stop reconcile if multiple ReleasePlanAdmissions exist", func() {
+		It("should stop reconcile if validateProcessingResources marks the release invalid", func() {
```

Comment:
It sucks that we have to check this here. Maybe a ticket to think about how to remove this from here (and add a proper validator) in the future?

Reply:
https://issues.redhat.com/browse/HACBS-2580